terraform/state.go

```diff
@@ -860,6 +863,12 @@ func (i *InstanceState) deepcopy() *InstanceState {
 			n.Attributes[k] = v
 		}
 	}
+	if i.Meta != nil {
+		n.Meta = make(map[string]string, len(i.Meta))
+		for k, v := range i.Meta {
+			n.Meta[k] = v
+		}
+	}
```

Comment:
Neither of these are strictly necessary, but I did them while trying to find the bug and they seemed like worth leaving in.